examples/api/lib/widgets/overlay/overlay.0.dart

```diff
@@ -0,0 +1,244 @@
+    removeHighlightOverlay();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Overlay Sample'),
+      ),
+      bottomNavigationBar: NavigationBar(
+        selectedIndex: currentPageIndex,
+        destinations: const <NavigationDestination>[
+          NavigationDestination(
+            icon: Icon(Icons.explore),
+            label: 'Explore',
+          ),
+          NavigationDestination(
+            icon: Icon(Icons.commute),
+            label: 'Commute',
+          ),
+          NavigationDestination(
+            selectedIcon: Icon(Icons.bookmark),
+            icon: Icon(Icons.bookmark_border),
+            label: 'Saved',
+          ),
+        ],
+      ),
+      body: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          Text(
+            'Use Overlay to highlight a NavigationBar destination',
+            style: Theme.of(context).textTheme.bodyMedium,
+          ),
+          const SizedBox(height: 20.0),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              // This creates a highlight Overlay for
+              // the Explore item.
+              ElevatedButton(
+                onPressed: () {
+                  createHighlightOverlay(
+                    alignment: Alignment.bottomLeft,
+                    borderColor: Colors.red,
+                  );
+                  setState(() {
+                    currentPageIndex = 0;
+                  });
+                },
+                child: const Text('Explore'),
+              ),
+              const SizedBox(width: 20.0),
+              // This creates a highlight Overlay for
+              // the Commute item.
+              ElevatedButton(
+                onPressed: () {
+                  createHighlightOverlay(
+                    alignment: Alignment.bottomCenter,
+                    borderColor: Colors.green,
+                  );
+                  setState(() {
+                    currentPageIndex = 1;
+                  });
+                },
+                child: const Text('Commute'),
+              ),
+              const SizedBox(width: 20.0),
+              // This creates a highlight Overlay for
+              // the Saved item.
+              ElevatedButton(
+                onPressed: () {
+                  createHighlightOverlay(
+                    alignment: Alignment.bottomRight,
+                    borderColor: Colors.orange,
+                  );
+                  setState(() {
+                    currentPageIndex = 2;
+                  });
+                },
+                child: const Text('Saved'),
+              ),
+            ],
+          ),
+          const SizedBox(height: 10.0),
+          ElevatedButton(
+            onPressed: () {
+              setState(() {
```

Comment:
no need for `setState`. From the code it looks like this widget itself isn't stateful, the overlay is?

Reply:
Removed